src/dialogs/alwaysonbotDialogs/UpdateProfile/UpdateAddress/getAddressesDialog.ts

```diff
@@ -0,0 +1,294 @@
+    
+    private async finalStep(stepContext: WaterfallStepContext<AddressDetails>): Promise<DialogTurnResult> {
+       
+       const recognizer = LUISAOSetup(stepContext);
+       const recognizerResult = await recognizer.recognize(stepContext.context);
+       const intent = LuisRecognizer.topIntent(recognizerResult, 'None', 0.7);
+       switch (intent) {
+           case 'Yes':
+              await stepContext.context.sendActivity(i18n.__('UpdateAddress'));
+               return await stepContext.beginDialog(CONTINUE_AND_FEEDBACK_DIALOG_STEP, ContinueAndFeedbackDialog);
+           case 'No':
+              return await stepContext.beginDialog(UPDATE_ADDRESS_DIALOG_STEP,UpdateAddressDialog);
+            default :
+            return stepContext.beginDialog(CALL_BACK_DAILOG_STEP, CallBackDailog);
+            return await stepContext.endDialog();
+
+       }
+   }
+    private GetEditedResponse(response:string,postalCode:AddressDetails)
+    {
+        if(postalCode.PostalCode!=null)
+        {
+        response=response.replace("@Postal_Code",postalCode.FullAddress);
+        response=response.replace("null","").replace("Null","").replace("Po","") ;
+        response=response.replace(","," ");
+        }
+        else
+        {
+            response=response.replace("@Postal_Code","");
+            response=response.replace("null","");
+            response=response.replace(","," ");
+        }
+        if(postalCode.UnitNumber!=null)
+        {
+            response=response.replace("#Unit_Number","#"+postalCode.UnitNumber);
+        }
+        else
+        {
+            response=response.replace("#Unit_Number","");
+        }
+        
+        return response;
+    }
+    private GetEditedChoices(response: string[], postalCode: string) {
+        let finaloptions=new Array();
+        response.forEach(element=>{
+            element=element.replace('@Postal_Code',postalCode)
+            finaloptions.push(element)
+        })
+        return finaloptions;
+    }
+    private GetSentence(word: string){
+        
+            var sentence = word.toLowerCase().split(" ");
+            let outSentence:string = "";
+            for(var i = 0; i< sentence.length; i++){
+               outSentence = outSentence + sentence[i][0].toUpperCase() + sentence[i].slice(1)+ " ";
+            }
+         return outSentence;
+        
+    }
+ 
+    private async getAddress(postalCode:string,langAPI:string): Promise<object> {
+        const query = {
+            AddressPostalCode: postalCode,
+         };
+        const url = new URL('https://services-nonprd.api.esdc-edsc.canada.ca/foundation/address/wsaddress/ground/v1/CAN/search');
+        url.search = new URLSearchParams(query).toString();
+        const headers = {
+            "Ocp-Apim-Subscription-Key": i18n.__('subscriptionKey'),
+            "Accept-Language": langAPI,
+        };
+        const response = await fetch( url.toString(), {headers} );
+        return await response.json();
```

Comment:
not sure we need await in the return.but I may be wrong

Reply:
yes we need this , this await is used for API response